jscomp/test/js_re_test.ml

```diff
@@ -1,12 +1,30 @@
 
 let suites = Mt.[
+  "matches", (fun _ ->
+    let re = [%re "/(\\d+)-(?:(\\d+))?/g"] in
+    let str = "3-" in
+    match re |> Js.Re.exec str with
+      | Some result ->
+        let defined = ref "" in
+        let _ = Js.Nullable.iter (Js.Re.matches result).(1) ((fun m ->
```

Comment:
I don't think you really need to use `iter` here at all. You should be able to just do

```ml
let defined = (Js.Re.matches result).(1) in
let undefined = ... in
Eq((Js.Nullable.return "3", Js.Nullable.null), (defined, undefined))
```